ibc-testkit/src/testapp/ibc/core/core_ctx.rs

```diff
@@ -48,7 +48,12 @@ where
     fn host_timestamp(&self) -> Result<Timestamp, HostError> {
         let host_height = self.host_height()?;
         let host_cons_state = self.host_consensus_state(&host_height)?;
-        Ok(host_cons_state.timestamp())
+        let timestamp = host_cons_state
+            .timestamp()
+            .map_err(|e| HostError::InvalidState {
```

Comment:
we should implement the associated methods for `HostError` such as

```rs
impl HostError {
  fn invalid_state(description: impl ToString) -> Self {
    Self::InvalidSate { description: description.to_string() }
  }
}
```

and then, use it here as

```rs
.map_err(HostError::invalid_state)
```

Reply:
Oh, slipped my mind we've added these methods. Ty. [4f11258](https://github.com/cosmos/ibc-rs/pull/1353/commits/4f112581c4155194e8830db65f85e14355e6db03)